### DISCUSSION TITLE
accessing to store data inside callback function

### DISCUSSION THREAD

**Discussion by tiger360** [alpinejs/alpine#2973] [1. Help] [answered]
Hello,
I wanted to access to "this" inside of a callback function but without success 
I did instead : 
`Alpine.store('darkMode').data  = records`
Is it the right way ?
`document.addEventListener('alpine:init', () => {
      Alpine.store('darkMode', {
        on: false,
        data : [{}],
        toggle(){
            this.on = ! this.on;
        },
        init() {
          grist.ready();
          grist.onRecords(function (records) {
            Alpine.store('darkMode').data = records;
          })
        }
      })
    })`

**Comment by SimoTod** [comment, marked as answer]
You must use an arrow function otherwise it will redefine `this`

**Reply by SimoTod** [reply]
`grist.onRecords((records) => { this.data = records })`

**Reply by tiger360** [reply]
omg it's working, thanks a lot 🥰🥰
